### ISSUE TITLE
Homing switches that are not at axis limits

### ISSUE THREAD

**Issue by T3P3** [dc42/RepRapFirmware#715]
Possibly, extend M208 to allow 3 values per axis?

https://forum.duet3d.com/post/233004

work around is to use G92 in the homing file to set the non zero position after homing.

**Comment by Aquilux** [comment]
I'd like to second this. I'd like to also note it'd be useful if the implementation was similar in format to M584. I imagine it'd help reduce confusion if you could have something like:

M584 X(A:B:C...) Y(A:B:C...) Z(A:B:C...)
M208 X(A:B:C...) Y(A:B:C...) Z(A:B:C...) S0
M208 X(a: b: c...) Y(a: b: c...) Z(a: b: c...) S1
where each entry corresponds to the same axis in sequence the same way end stop configuration does it.

I'm not sure how to deal with backwards compatibility with the variant format available or alternatively what characters to use to delineate between entries in the case of the alternate M208 behavior though.

**Comment by dc42** [comment]
I agree with extending M208 to support 3 values per axis. The 3rd value would be the homing switch position.

**Comment by T3P3** [comment]
when implementing this we need to consider this request as well:
Different endstop position for each driver in an axis : https://github.com/Duet3D/RepRapFirmware/issues/627

**Comment by Aquilux** [comment]
Just to clarify, because I might be misunderstanding your replies: My suggestion was to have M208's behavior align with the behavior of M92, M201, M203, M350, M556, and M906 as depicted in [this guide for multiple motor driver configuration.](https://docs.duet3d.com/User_manual/Connecting_hardware/Motors_configuring) As it is now it stands apart as wholly divergent behavior in comparison to all the other commands intended for defining and configuring hardware and assigning it to specified axes.

**Comment by T3P3** [comment]
Forum thread: https://forum.duet3d.com/topic/33961
